Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs

```diff
@@ -0,0 +1,750 @@
+        ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitActualTypeArgs(List<Type> typeArgs, IToken tok, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override string GenerateLhsDecl(string target, Type type, ConcreteSyntaxTree wr, IToken tok) {
+      return target;
+    }
+
+    protected override void EmitPrintStmt(ConcreteSyntaxTree wr, Expression arg) {
+      var wStmts = wr.Fork();
+      wr.Write("print(");
+      EmitToString(wr, arg, wStmts);
+      wr.WriteLine(");");
+    }
+
+    private void EmitToString(ConcreteSyntaxTree wr, Expression arg, ConcreteSyntaxTree wStmts) {
+      TrExpr(arg, wr, false, wStmts);
+    }
+
+    protected override void EmitReturn(List<Formal> outParams, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree CreateLabeledCode(string label, bool createContinueLabel, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitBreak(string label, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitContinue(string label, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitYield(ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.Iterators);
+    }
+
+    protected override void EmitAbsurd(string message, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitHalt(IToken tok, Expression messageExpr, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree EmitIf(out ConcreteSyntaxTree guardWriter, bool hasElse, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree EmitBlock(ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree EmitForStmt(IToken tok, IVariable loopIndex, bool goingUp, string endVarName,
+      List<Statement> body, LList<Label> labels, ConcreteSyntaxTree wr) {
+      // var direction = goingUp ? "to" : "downto";
+      // var lowWr = new ConcreteSyntaxTree();
+      // wr.Format($"for {IdName(loopIndex)} := {lowWr} {direction} {endVarName}");
+      // var bodyWr = wr.NewBlock();
+      // TrStmtList(body, bodyWr);
+      // return lowWr;
```

Comment:
This is commented out because it's not working well enough? You can keep it, just asking.

Reply:
Correct, and I would like to keep it because it is going in the right direction and is an example, for me, on how to use the "write head" properly